@deedrah, I think I just used:

```
$ nix-shell -I nixpkgs=https://github.com/NixOS/nixpkgs/archive/86c5bfcdc610b06d4c943de9438148cb422e528e.tar.gz -p gnome3.evolution
```

… but with the previous commit SHA1.

But now when I tried that, with the current one:

```
builder for ‘/nix/store/nqirmq896dyngcj4ir1c4x9pllcb3han-liboauth-1.0.3’ failed previously (cached)
```

The error was:

```
/nix/store/xmfz6cv13ippmg425rar74nygmlj06jl-bash-4.4-p12/bin/bash ../libtool  --tag=CC   --mode=compile gcc -DHAVE_CON
FIG_H -I.    -Wall -I/nix/store/agdpcdar9fmjcd4cm3algmjzpxsa5bny-nss-3.32.1/include/nss -I/nix/store/smhc7vh4a57j08j9k
nrc73v6vpnybmj9-nspr-4.16-dev/include  -g -O2 -c -o liboauth_la-oauth.lo `test -f 'oauth.c' || echo './'`oauth.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -Wall -I/nix/store/agdpcdar9fmjcd4cm3algmjzpxsa5bny-nss-3.32.1/include/nss 
-I/nix/store/smhc7vh4a57j08j9knrc73v6vpnybmj9-nspr-4.16-dev/include -g -O2 -c oauth.c  -fPIC -DPIC -o .libs/liboauth_l
a-oauth.o

No such file or directory
 #  include "pk11pub.h"
                       ^
compilation terminated.
make[2]: *** [Makefile:377: liboauth_la-oauth.lo] Error 1

```

So this is unrelated to your PR. Could you give me a command that works for you w/ gedit?

Thank you for your effort! ♥

--- a/pkgs/development/libraries/aspell/aspell-with-dicts.nix
+++ b/pkgs/development/libraries/aspell/aspell-with-dicts.nix
@@ -1,5 +1,9 @@
 # Create a derivation that contains aspell and selected dictionaries.
 # Composition is done using `pkgs.buildEnv`.
+# Beware of that `ASPELL_CONF` used by this derivation is not always
+# respected by libaspell (#28815) and in some cases, when used as
+# dependency by another derivation, the passed dictionaries will be
+# missing. However, invoking aspell directly should be fine.
 
 { aspell
 , aspellDicts

--- a/pkgs/development/libraries/aspell/data-dirs-from-nix-profiles.patch
+++ b/pkgs/development/libraries/aspell/data-dirs-from-nix-profiles.patch
@@ -1,0 +1,38 @@
+diff --git a/common/info.cpp b/common/info.cpp
+index 8291cc7..6216326 100644
+--- a/common/info.cpp
++++ b/common/info.cpp
+@@ -36,6 +36,7 @@
+ #include "strtonum.hpp"
+ #include "lock.hpp"
+ #include "string_map.hpp"
++#include "file_util.hpp"
+ 
+ #include "gettext.h"
+ 
+@@ -495,6 +496,25 @@ namespace acommon {
+     lst.clear();
+     lst.add(config->retrieve("data-dir"));
+     lst.add(config->retrieve("dict-dir"));
++    if (config->lookup("data-dir") == NULL && config->lookup("dict-dir") == NULL) {
++        const char* cprofiles = getenv("NIX_PROFILES");
++        if (cprofiles != NULL) {
++            char* profiles = strdup(cprofiles);
++            char* profile = profiles;
++            char* end = profile;
++            while (*end != '\0') {
++                if (*end == ' ') {
++                    *end = '\0';
++                    lst.add(add_possible_dir(profile, "lib/aspell"));
++                    profile = ++end;
++                } else {
++                    ++end;
++                }
++            }
++            lst.add(add_possible_dir(profile, "lib/aspell"));
++            free(profiles);
++        }
++    }
+   }
+ 
+   DictExt::DictExt(ModuleInfo * m, const char * e)

--- a/pkgs/development/libraries/aspell/default.nix
+++ b/pkgs/development/libraries/aspell/default.nix
@@ -1,4 +1,5 @@
-{stdenv, fetchurl, perl}:
+{stdenv, fetchurl, perl
+, searchNixProfiles ? true}:
 
 stdenv.mkDerivation rec {
   name = "aspell-0.60.6.1";
@@ -10,6 +11,8 @@ stdenv.mkDerivation rec {
 
   patchPhase = ''
     patch interfaces/cc/aspell.h < ${./clang.patch}
+  '' + stdenv.lib.optionalString searchNixProfiles ''
+    patch -p1 < ${./data-dirs-from-nix-profiles.patch}
   '';
 
   buildInputs = [ perl ];
@@ -21,28 +24,6 @@ stdenv.mkDerivation rec {
       --enable-pkglibdir=$out/lib/aspell
       --enable-pkgdatadir=$out/lib/aspell
     );
-  '';
-
-  postInstall = ''
-    local prog="$out/bin/aspell"
-    local hidden="$out/bin/.aspell-wrapped"
-    mv "$prog" "$hidden"
-    cat > "$prog" <<END
-    #! $SHELL -e
-    if [ -z "\$ASPELL_CONF" ]; then
-      for p in \$NIX_PROFILES; do
-        if [ -d "\$p/lib/aspell" ]; then
-          ASPELL_CONF="data-dir \$p/lib/aspell"
-        fi
-      done
-      if [ -z "\$ASPELL_CONF" ] && [ -d "\$HOME/.nix-profile/lib/aspell" ]; then
-        ASPELL_CONF="data-dir \$HOME/.nix-profile/lib/aspell"
-      fi
-      export ASPELL_CONF
-    fi
-    exec "$hidden" "\$@"
-    END
-    chmod +x "$prog"
   '';
 
   meta = {

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -7769,7 +7769,9 @@ with pkgs;
 
   aspellDicts = recurseIntoAttrs (callPackages ../development/libraries/aspell/dictionaries.nix {});
 
-  aspellWithDicts = callPackage ../development/libraries/aspell/aspell-with-dicts.nix { };
+  aspellWithDicts = callPackage ../development/libraries/aspell/aspell-with-dicts.nix {
+    aspell = aspell.override { searchNixProfiles = false; };
+  };
 
   attica = callPackage ../development/libraries/attica { };
 

